pandas/core/tools/datetimes.py

```diff
@@ -734,7 +736,17 @@ def to_datetime(
             convert_listlike = partial(convert_listlike, name=arg.name)
             result = convert_listlike(arg, format)
     elif is_list_like(arg):
-        cache_array = _maybe_cache(arg, format, cache, convert_listlike)
+        try:
+            cache_array = _maybe_cache(arg, format, cache, convert_listlike)
+        except tslibs.OutOfBoundsDatetime:
```

Comment:
I *think* this is the best we can do here, at least without major surgery. It'll involve double-parsing the input for large arrays with `errors='coerce'`.

Reply:
ok, pls create an issue as this should be handled directly in _maybe_cache itself